frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -365,7 +365,9 @@ export const transformTopologyData = (
 
   const knSvcResources: K8sResourceKind[] = _.get(resources, ['ksservices', 'data'], []);
   knSvcResources.length && getKnativeTopologyData(knSvcResources, NodeType.KnService);
-  const knEventSources: K8sResourceKind[] = getKnativeEventSources();
+  const knEventSources: K8sResourceKind[] = filters.display.eventSources
```

Comment:
you might want to write it like :
```suggestion
  const knEventSources: K8sResourceKind[] = filters.display.eventSources && getKnativeEventSources();
```

Reply:
I caution against this change. `&&` is a sneaky beast in JavaScript. I am unsure if TypeScript properly catches this, but if your left-hand variable is falsey, you get that value. This isn't so much of a problem when rendering React as falsey usually means don't render it.

However if you do it with a JS statement, you can be getting unexpected values into your variable:
```
1 && 2
// 2
0 && 1
// 0
true && []
// []
false && []
// false
undefined && ['hello', 'world']
// undefined
```

Which can be confusing for your variable (in this case `knEventSources`). Especially if you want it to be an array.

Reply:
@andrewballantyne Thank you for this piece of information. 
When I executed this PR locally (after pulling in #3476) it worked fine without an error so I went ahead and suggested the change. But I will make a note of the example you just shared